compiler/compiler.py

```diff
@@ -194,23 +191,24 @@ def get_task(src_id: str, chk: str, fmt: Format = Format.PDF) -> Task:
     task_id = _get_task_id(src_id, chk, fmt)
     result = do_compile.AsyncResult(task_id)
     stat = Status.IN_PROGRESS
-    owner: Optional[str] = None
     reason = Reason.NONE
+    _info: Dict[str, str]
     if result.status == 'PENDING':
         raise NoSuchTask(f'No such task: {task_id}')
```

Comment:
The PENDING state means different things but I'm surprised to see it triggers NoSuchTask exception. Is this like a failed launch attempt or halted state? Not really part of this PR. I suspect the meaning of this state evolved over time. Pending gives me the impression the job is waiting and will eventually run. No biggie.

Reply:
It's a foible of Celery. When you instantiate an ``AsyncResult``, it will wait lazily to actually retrieve anything from redis until you access its properties, and show a ``PENDING`` state (even if there is no such task). And in some cases (maybe this is fixed now), but you could end up just waiting+blocking on non-existant tasks.

So what I generally do is set the state to something other than ``PENDING`` when a task is created. ``SENT`` in this case, IIRC. That way if we see ``PENDING`` we know for sure that the task doesn't actually exist.